src/app/modules/payments/stripe.utils.ts

```diff
@@ -435,11 +476,25 @@ export const convertToProofOfPaymentFormat = (
       '<td class="st-Spacer st-Spacer--gutter" width="48"></td>',
     )
 
+  // Check if payment by products (instead of variable / fixed payments) is used. If yes, modify proof of payment
+  const paymentByProductsEdits =
+    products.length === 0
+      ? commonEdits
+      : commonEdits
+          .replace(
+            /<td class="Table-description Font Font--body".+">\n.+test\n.+<\/td>/,
+            `${paymentByProductsDescription}`,
+          )
+          .replace(
+            /<td class="Table-amount Font Font--body".+">\n.+\n.+<\/td>/,
+            `${paymentByProductsPrice}`,
+          )
+
   // If GST is applicable, 'Invoice' is used instead of 'Receipt'
   const gstDependentEdits = gstApplicable
-    ? commonEdits
+    ? paymentByProductsEdits
         .replace(/(<title>.*?)receipt(.*?<\/title>)/, '$1invoice$2')
-        .replace(/Receipt from /g, 'Invoice from ')
+        .replace(/Receipt from /g, 'Invoice from')
```

Comment:
I notice that an empty space was removed here, is it intended?

Reply:
good spot! not intended, fixed in [d00030f](https://github.com/opengovsg/FormSG/pull/6574/commits/d00030faa17fcb6fb88f28ec126998d633569e3d)